sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/namedExpressions.scala

```diff
@@ -112,6 +112,8 @@ case class Alias(child: Expression, name: String)(
   extends NamedExpression with trees.UnaryNode[Expression] {
 
   override type EvaluatedType = Any
+  // Alias(Generator, xx) need to be transformed into Generate(generator, ...)
+  override lazy val resolved = childrenResolved && !child.isInstanceOf[Generator]
```

Comment:
@marmbrus sorry, I am not so sure your mean, most of change that I made is move the `aliases` from `Generator` to `Generate`, which means `Generator` developers will not cares about the aliases any more.
The `Alias(Generator, xx)` here will be transformed to `Generate(generator, aliases)`, that's why I marked it as `unresolved` for Alias.